packages/wrapped-keys/src/lib/utils.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  AccessControlConditions,
+  AuthSig,
+  SessionKeySignedMessage,
+  SessionSigsMap,
+} from '@lit-protocol/types';
+import { log } from '@lit-protocol/misc';
+import { CHAIN_ETHEREUM } from './constants';
+import { ethers } from 'ethers';
+// import { log } from 'console';
+
+export function getFirstSessionSig(pkpSessionSigs: SessionSigsMap): AuthSig {
+  const keys = Object.keys(pkpSessionSigs);
+  if (keys.length === 0) {
+    throw new Error(
+      `Invalid pkpSessionSigs, length zero: ${JSON.stringify(pkpSessionSigs)}`
+    );
+  }
+
+  const firstSessionSig: AuthSig = pkpSessionSigs[keys[0]];
+  log(`Session Sig being used: ${JSON.stringify(firstSessionSig)}`);
+
+  return firstSessionSig;
+}
+
+export function getPkpAddressFromSessionSig(pkpSessionSig: AuthSig): string {
+  const sessionSignedMessage: SessionKeySignedMessage = JSON.parse(
+    pkpSessionSig.signedMessage
+  );
+
+  const capabilities = sessionSignedMessage.capabilities;
+  if (capabilities.length !== 1) {
```

Comment:
Why do we wish to restrict the capabilities to a single item?

Reply:
Since this is on Cayenne there has to be only 1 BLS innerAuthSig. But for Habanero & Manzano we need another one for payment delegation so the Lambda need to be updated for it